src/main/java/org/gaul/s3proxy/junit/S3ProxyExtension.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2014-2020 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy.junit;
+
+import java.net.URI;
+
+import org.gaul.s3proxy.AuthenticationType;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * A JUnit 5 Extension that manages an S3Proxy instance which tests
+ * can use as an S3 API endpoint.
+ */
+public final class S3ProxyExtension
+        implements AfterAllCallback, BeforeAllCallback {
+
+    private final S3ProxyJunitCore core;
+
+    public static final class Builder {
+
+        private final S3ProxyJunitCore.Builder builder;
+
+        private Builder() {
+            builder = new S3ProxyJunitCore.Builder();
+        }
+
+        public Builder withCredentials(AuthenticationType authType,
+                                       String accessKey, String secretKey) {
+            builder.withCredentials(authType, accessKey, secretKey);
+            return this;
+        }
+
+        public Builder withCredentials(String accessKey, String secretKey) {
+            return withCredentials(AuthenticationType.AWS_V2_OR_V4, accessKey,
+                    secretKey);
```

Comment:
```suggestion
          builder.withCredentials(accessKey, secretKey);
          return this;
```
I think this might be better to keep it consistent and to keep the logic of defaulting to `AuthenticationType.AWS_V2_OR_V4` in one place.